### DISCUSSION TITLE
sx vs css prop?

### DISCUSSION THREAD

**Discussion by testerez** [mantinedev/mantine#2339] [Q&A] [answered]
What is the difference between the `sx` prop and the emotion `css` prop?

It looks like both work the same way and produce the same result. The only difference I noticed is that when using `css`, the component name is added to the produced css classname. It's not the case with `sx`. Also, `css` prop works with any component, so I'd rather use the same thing everywhere. But is there any reason why I should use `sx` instead?

**Comment by rtivital** [comment, marked as answer]
- `css` prop does not support Mantine theme, you won't be able to use function syntax like in sx: `css={(theme) => ({})}`
- `css` prop requires additional jsx transform (usually though a babel plugin)
- styles added with `css` will not use Mantine emotion cache unless your wrap your app with CacheProvider – this will have an impact if you are using custom emotion cache option, for example for RTL configuration.

If you want to add styles to any non-Mantine component, then you have several options:
- Use Box instead of this element/component: `<Box component="p" />`, `<Box component={Link} />`
- Add styles via createStyles function
- Use emotion/styled
- Use your own styling solution – in this case you will lose access to Mantine theme

**Reply by testerez** [reply]
Thanks for the great and quick answer @rtivital ! 🙏
